### ISSUE TITLE
Allow components be mountable on multiple frames / bikes

### ISSUE THREAD

**Issue by OleksiyRudenko** [Gerben321/bike-care-tracker#68]
**Is your feature request related to a problem? Please describe.**
I may want to use clipless pedals on multiple bikes.

**Describe the solution you'd like**
Allow m:n relation between components and frames/bikes.

**Describe alternatives you've considered**
none

**Additional context**
I do not have this particular situation as I own only one frame.
Although I could use some easily mountable components (e.g. pedals, lights, saddle) with rented bikes.
Guess this feature request is worth verification with 2+ frames owners.

**Comment by Gerben321** [comment]
I feared this request would come, haha. I have multiple bikes but I do not share things (apart from lights and water bottles and stuff like that). Right now a component is strictly connected to a bike. This means a component has a link to a bike directly.

In order to facilitate this would mean that a component is no longer directly connected to a bike, but has a connection to a bike through the 'usage period' where we now select a start date, and an optional end date. There would also be the option for the bike a that moment.

This is a major change, not only because of that structure, but also because a component can be 'roaming' freely without a bike, but it still has to be visible somewhere. So we then have to add some list of components etc. 

It's doable for sure, but it's a significant change. I am willing to do it, but not if we compromise the ease of use that we have right now, or if we can think of a nice hybrid solution where we keep all the current screens and ways of adding/editing things like a component is still part of a bike, but with the option to move it to another bike, and see a list of components like you have a list of bikes.

So you can see the current list of components when you go to a bike, but also have a COMPLETE list of ALL components in a separate list where you can see if it's installed, and on which bike etc.

Anyway, we have to think this one through because it has consequences. For the UI/UX and data structure.

**Comment by OleksiyRudenko** [comment]
I do not insist on this feature (do not have a use case for this).
I think that even if it gets discarded it would be helpful to document CBA, just in case the idea arises again, so the initiator could be redicrected to already collected arguments pro and con. :)

Anyway, feel free de-priorotizing it.

**Comment by Gerben321** [comment]
Good point. I will put this as a nice-to-have right now for the current user-base. But this will get important if we ever get more users, so good to do when things are calm and I have some time to spend for sure!

**Comment by Konstantin-Levin** [comment]
Yes this will come up for sure as it is quite a common practice. I don’t require it myself, but I know plenty of people who frequently swap pedals and wheelsets etc between bikes. 
It will unfortunately add complication to the app as you say but maybe there is a way to do it simply. 
   
Of course things like wheels have the added complication of having cassettes and tyres on them 😬 
So if you did decide to do this in the future, a way to move multiple components at once (with checkboxes or whatever) would probably help.

**Comment by Gerben321** [comment]
I think it could maybe help to have groups of components. So a group consists of the rear wheel, cassette, and tire. And you move it together.

**Comment by Gerben321** [comment]
On-topic: I've got a first setup for the feature to group components and swap them between bikes.

Overview of groups:
<img width="972" height="207" alt="Image" src="https://github.com/user-attachments/assets/d589306f-05ec-438f-a8ff-18165d41a806" />
Editing/creating groups:
<img width="661" height="251" alt="Image" src="https://github.com/user-attachments/assets/2ccfdb34-f35a-4118-ac66-727bef41a776" />
Transfer to another bike:
<img width="505" height="159" alt="Image" src="https://github.com/user-attachments/assets/59fa63ae-c0e8-47f9-b63c-e2ad35d30e58" />
